dauphine-open-data-lib/src/main/java/io/github/oliviercailloux/y2018/opendata/entity/Person.java

```diff
@@ -0,0 +1,264 @@
+ * Person is either Student or Teacher. In this class you have basic attributes to
+ * manipulate class Person and getters an setters
+ * <br />
+ * @author Gandi Taric
+ */
+
+@Entity
+public class Person extends AbstractEntity {
+
+	private static final long serialVersionUID = -4768931293875213592L;
+
+	
+	
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public Person() {
+		super();
+		accessServices= new ArrayList();
+		personnalMail = new ArrayList();
+		dauphineMail  = new ArrayList();	
+	}
+
+	@Override
+	public Long getId() {
+		return id;
+	}
+
+	public String getFirstName() {
+
+		return firstName;
+	}
+
+	public void setFirstName(String firstName) {
+
+		this.firstName = firstName;
+	}
+
+	public String getLastName() {
+
+		return LastName;
+	}
+
+	public void setLastName(String lastName) {
+
+		LastName = lastName;
+	}
+
+	public String getAccountType() {
+
+		return accountType;
+	}
+
+	public void setAccountType(String accountType) {
+
+		this.accountType = accountType;
+
+	}
+
+	public List<String> getMemberShip() {
+
+		return memberShip;
+
+	}
+
+	public void setMemberShip(List<String> memberShip) {
+
+		this.memberShip = memberShip;
+
+	}
+	
+	public  List<String> getDauphineMail() {
+
+		return Collections.unmodifiableList(dauphineMail);
+
+	}
+
+	public void setDauphineMail(String dauphineMail) {
+
+		this.dauphineMail.add(dauphineMail);
+
+	}
+
+	public String getOffice() {
+
+		return office;
+	}
+
+	/**
+	 * @param office if this parameter is null, it will be convert to an empty
+	 *               string
+	 */
+	public void setOffice(String office) {
+
+		this.office = Strings.nullToEmpty(office);
+	}
+
+	/**
+	 * 
+	 * @return a list of services accesses of the person
+	 */
+	public List<String> getAccessServices() {
+
+		return Collections.unmodifiableList(accessServices);
+	}
+
+	public void setAccessServices(String service) {
+
+		this.accessServices.add(service);
+	}
+
+	public String getPhoneNumer() {
+
+		return phoneNumer;
+	}
+
+	/**
+	 * @param phoneNumer if this parameter is null, it will be remplaced by an empty
+	 *                   string
+	 */
+	public void setPhoneNumer(String phoneNumer) {
+
+		this.phoneNumer = Strings.nullToEmpty(phoneNumer);
+	}
+
+	public String getTraining() {
+
+		return training;
+	}
+
+	/**
+	 * 
+	 * @param training will remplaced by empty string if it's null
+	 */
+	public void setTraining(String training) {
+
+		this.training = Strings.nullToEmpty(training);
+
+	}
+
+	public String getFax() {
+
+		return fax;
+	}
+
+	public void setFax(String fax) {
+
+		this.fax = Strings.nullToEmpty(fax);
+	}
+
+	public List<String> getPersonnalMail() {
+
+		return Collections.unmodifiableList(personnalMail);
+	}
+
+	public void setPersonnalMail(String personnalMail) {
+
+		this.personnalMail.add(personnalMail);
+	}
+
+	public String getIne() {
+
+		return ine;
+	}
+
+	public void setIne(String ine) {
+
+		this.ine = Strings.nullToEmpty(ine);
```

Comment:
Bonne protection. Commenter dans la classe que les strings renvoyés par les getteurs ne sont jamais nulls, si c’est le cas. Ou commenter chaque méthode au cas par cas.

Reply:
Ok bien noté

Reply:
Le code ne fait pas ce que le commentaire laisse penser. Il faut refuser un Ine sur un non-étudiant, s’il y a un autre moyen de savoir si une personne est un étudiant.
Si une personne est un étudiant par définition lorsqu’elle a un Ine, alors le commentaire (sur le getter et sur le setter) doit l’indiquer comme ça.